HomeWork2/DmytroFenevHW2.cpp

```diff
@@ -0,0 +1,159 @@
+#include <cmath>
+#include <iostream>
+#include <string>
+
+int main() {
+  int z;
+  std::cout << "Zadacha №1, N2, N3, N4, N5\n";
+  std::cin >> z;
+  int x, xp, xp1;
+  int y, y1, yp;
+  int n, np, nc;
+  int sum, sum1, sum2;
+  int l2;
+  int k2;
+  double k3;
+  int sample[10] = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
+  const int Min = 1;
+  const int Max = 50;
+  int nElement = 0;
+  int chislo;
+  const int cmin = -60;
+  const int cmax = 90;
+  switch (z) {
+  case 1:
+    n = 1; //число цифр
+    std::cout << "\tEnter an integer" << std::endl;
+    std::cin >> x;
+    xp = x; //сохранение х после подсчета количества цифр
+    while ((x /= 10) > 0)
+      n++;
+    std::cout << "\tNumber of digits=" << n << std::endl;
+    y = xp / 10;
+    y1 = xp % 10; // number of low order
+    sum = y1;
+    for (int ii = 1; ii < n; ii++) {
+      yp = y / 10;
+      y1 = y % 10;
+      sum = sum + y1; //сумма цифр
+      y = yp;
+    }
+    std::cout << "\tsum=" << sum << std::endl;
+    std::cout << "\tAverage number =" << sum / n << std::endl;
+    break;
+  case 2:
+    n = 6; //число цифр
+    np = 1;
+    std::cout << "\tEnter six digits" << std::endl;
+    std::cin >> x;
+    y = x; //сохранение х после подсчета количества цифр
+    while ((x /= 10) > 0)
+      np++;
+    std::cout << np << std::endl;
+    while ((np < n) || (np > n)) {
+      np = 1;
+      std::cout << "\tEnter not six  digits" << std::endl;
+      std::cin >> x;
+      y = x;
+      while ((x /= 10) > 0)
+        np++;
+    }
+    sum1 = 0;
+    sum2 = 0;
+    for (int ii = 1; ii <= n; ii++) {
+      if (ii < 4) {
+        yp = y / 10;
+        y1 = y % 10;
+        sum1 = sum1 + y1;
+        y = yp;
```

Comment:
```suggestion
        sum1 += (y % 10);
        y /= 10;
```
избегайте в коде избыточных переменных, ваш код становится чистым в плане чтения кода

Reply:
я изменил
 int sum1, sum2;
        sum1 = 0;
        sum2 = 0;
        for (int ii = 0; ii < numberDigit;ii++)
        {
            if (ii < numberDigit / 2) 
             sum1 += x % 10; 
            else
                sum2 += x % 10;
            x = x / 10;
        }